fem/eltrans.hpp

```diff
@@ -356,12 +379,57 @@ class IntegrationPointTransformation
    void Transform (const IntegrationRule  &, IntegrationRule  &);
 };
 
-class FaceElementTransformations
+class FaceElementTransformations : public IsoparametricTransformation
 {
+private:
+   int mask;
+
+   IntegrationPoint eip1, eip2;
+
 public:
-   int Elem1No, Elem2No, FaceGeom;
-   ElementTransformation *Elem1, *Elem2, *Face;
+   int Elem1No, Elem2No;
+   Geometry::Type &FaceGeom; ///< @deprecated Use GetGeometryType instead
+   ElementTransformation *Elem1, *Elem2;
+   ElementTransformation *Face; ///< @deprecated No longer necessary
    IntegrationPointTransformation Loc1, Loc2;
+
+   FaceElementTransformations() : FaceGeom(geom), Face(this) {}
+
+   /** @brief Method to set the geometry type of the face.
+
+       @note This method should only be used when
+       [Par]Mesh::GetFaceTransformation will not be called i.e. when the
+       face transformation will not be needed but the neighboring
+       element transformations will be.
```

Comment:
This comment sounds too restrictive. I think the method can still be called even with `GetFaceTransformation`,  it would just overwrite the value. It is the caller's responsibility to set the correct value.

Reply:
True.  I can reword it.